### PR TITLE
LPD-34667

### PR DESCRIPTION
**Pull request by rijema**
Related issue: https://liferay.atlassian.net/browse/LPD-34667

### CODE REVIEW

**Comment by marcelabc** [comment]
I just wanted to let you know that I edited the description to include the "Related Issue" (with the Jira ticket). Please always add this information in your following reviews if you can!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -4986,7 +4986,8 @@ else if (StringUtil.equals(
 			if (!entryValueString.isEmpty()) {
 				long value = GetterUtil.getLong(entryValue);
 
-				if (!StringUtil.equals(
+				if ((value == 0) &&
+					!StringUtil.equals(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4990]: The same situation happens with `ObjectFieldConstants.DB_TYPE_INTEGER`. Can you apply your fix there and add a test for it, too?

Comment (rijema) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4991]: Done!  (๑ > ᴗ < ๑)

**Comment by rijema** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by rijema** [comment]
ci:test:object

**Comment by rijema** [comment]
All the erros on testray contains the info of Failed to run, so I'm restarting this analysis

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**